src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs

```diff
@@ -26,10 +36,236 @@ internal static class LightGbmShared
+            public int NumberOfIterations = Defaults.NumberOfIterations;
+
+            /// <summary>
+            /// The shrinkage rate for trees, used to prevent over-fitting.
+            /// </summary>
+            /// <value>
+            /// Valid range is (0,1].
+            /// </value>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Shrinkage rate for trees, used to prevent over-fitting. Range: (0,1].",
+                SortOrder = 2, ShortName = "lr", NullName = "<Auto>")]
+            [TGUI(Label = "Learning Rate", SuggestedSweeps = "0.025-0.4;log")]
+            [TlcModule.SweepableFloatParamAttribute("LearningRate", 0.025f, 0.4f, isLogScale: true)]
+            public double? LearningRate;
+
+            /// <summary>
+            /// The maximum number of leaves in one tree.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Maximum leaves for trees.",
+                SortOrder = 2, ShortName = "nl", NullName = "<Auto>")]
+            [TGUI(Description = "The maximum number of leaves per tree", SuggestedSweeps = "2-128;log;inc:4")]
+            [TlcModule.SweepableLongParamAttribute("NumLeaves", 2, 128, isLogScale: true, stepSize: 4)]
+            public int? NumberOfLeaves;
+
+            /// <summary>
+            /// The minimal number of data points required to form a new tree leaf.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Minimum number of instances needed in a child.",
+                SortOrder = 2, ShortName = "mil", NullName = "<Auto>")]
+            [TGUI(Label = "Min Documents In Leaves", SuggestedSweeps = "1,10,20,50 ")]
+            [TlcModule.SweepableDiscreteParamAttribute("MinDataPerLeaf", new object[] { 1, 10, 20, 50 })]
+            public int? MinimumExampleCountPerLeaf;
+
+            /// <summary>
+            /// The maximum number of bins that feature values will be bucketed in.
+            /// </summary>
+            /// <remarks>
+            /// The small number of bins may reduce training accuracy but may increase general power (deal with over-fitting).
+            /// </remarks>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Maximum number of bucket bin for features.", ShortName = "mb")]
+            public int MaximumBinCountPerFeature = 255;
+
+            /// <summary>
+            /// Determines which booster to use.
+            /// </summary>
+            /// <value>
+            /// Available boosters are <see cref="DartBooster"/>, <see cref="GossBooster"/>, and <see cref="GradientBooster"/>.
+            /// </value>
+            [Argument(ArgumentType.Multiple, HelpText = "Which booster to use, can be gbtree, gblinear or dart. gbtree and dart use tree based model while gblinear uses linear function.", SortOrder = 3)]
+            internal IBoosterParameterFactory Booster = new GradientBooster.Options();
```

Comment:
>Booster [](start = 46, length = 7)

You can name it `BoosterFactory` and rename `BoosterParameter` to `Booster`.
Just add `Name="Booster"` in ArgumentAttribute